src/commands/image/imageSource/containerRegistry/ContainerRegistryImageConfigureStep.ts

```diff
@@ -41,6 +41,10 @@ export class ContainerRegistryImageConfigureStep extends AzureWizardExecuteStep<
             }
         }
 
+        // Preserve existing secrets/registries even if new ones haven't been added
+        context.secrets ??= secrets;
+        context.registries ??= registries;
```

Comment:
This is really covering the else condition above (so third party public Docker registries), but it's also just a general good fallback to have here.